src/euroeval/benchmark_modules/vllm.py

```diff
@@ -1430,3 +1436,42 @@ def get_vllm_tokenisation_params(
         config_format=config_format,
         load_format=load_format,
     )
+
+
+def select_backend_and_parallelism() -> tuple[str, int, int]:
```

Comment:
[nit] We could consider caching this function (using the `functools.cache` decorator), to avoid unnecessary repeated checks.